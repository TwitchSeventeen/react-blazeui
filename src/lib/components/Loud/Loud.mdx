---
name: Loud
---

import { Playground, PropsTable } from 'docz'
import Loud from './index';

# Loud

This adds a loud emphasis to text. It can be used inline (ie it's not a block level element).

## Basic usage
<Playground>
    <Loud>
        Sometimes you want to shout
    </Loud>
    <br />
    Regular text
</Playground>

## Properties
<PropsTable of={Loud} />