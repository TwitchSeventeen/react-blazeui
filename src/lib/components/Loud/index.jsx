import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Loud(props) {
  const {
    children, className, dataName, ...other
  } = props;
  const classes = classNames('u-text--loud', className);
  return (
    <span
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </span>
  );
}

Loud.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
};

Loud.defaultProps = {
  children: null,
  className: null,
  dataName: 'loud',
};

export default Loud;
