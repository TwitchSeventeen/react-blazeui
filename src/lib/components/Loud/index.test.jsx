import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Loud from './index';

describe('Loud', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Loud).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Loud>
        lorem ipsum dolor
      </Loud>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
