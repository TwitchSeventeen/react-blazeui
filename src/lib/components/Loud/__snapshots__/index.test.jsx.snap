// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loud renders correctly 1`] = `
<DocumentFragment>
  <span
    class="u-text--loud"
    data-name="loud"
  >
    lorem ipsum dolor
  </span>
</DocumentFragment>
`;
